developers/academy/js/starter_multimodal_data/102_mm_collections/30_import_data.mdx

```diff
@@ -0,0 +1,109 @@
+---
+title: Import data
+description: Importing Data into Multimodal Data Collections
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import FilteredTextBlock from '@site/src/components/Documentation/FilteredTextBlock';
+import TSCode from '!!raw-loader!../_snippets/102_collection.mts';
+
+## <i class="fa-solid fa-code"></i> Code
+
+This example imports the movie data into our collection.
+
+<FilteredTextBlock
+  text={TSCode}
+  startMarker="// BatchImportData"
+  endMarker="// END BatchImportData"
+  language="ts"
+/>
+
+The code:
+- Loads the source text and image data
+- Gets the collection
+- Enters a context manager with a batcher (`batch`) object
+- Loops through the data and:
+    - Finds corresponding image to the text
+    - Converts the image to base64
+    - Adds objects to the batcher
+- Prints out any import errors
+
+## <i class="fa-solid fa-chalkboard"></i> Explain the code
+
+### <i class="fa-solid fa-chalkboard"></i> Preparation
+
+We use the native Node.js `fetch()` to load the data from the source, in this case a JSON file containing text data and a Zip file containing posters. The text data is then converted to a JSON object for easier manipulation and the images are extracted from the Zip file.
+
+Then, we create a collection object (with `client.collections.get`) so we can interact with the collection.
+
+### <i class="fa-solid fa-chalkboard"></i> Iterating over data
+
+The `for` loop is used in conjunction with `Object.keys()` to iterate through the elements in our JSON file. While iterating we increment the counter variable that lets us bulk insert objects in batches.
+
+<FilteredTextBlock
+  text={TSCode}
+  startMarker="// Iterate through data"
+  endMarker="// END Iterate through data"
+  language="ts"
+/>
+
+> We use `MAX_ITEMS` to limit the number of objects we import to our collection. Trail keys for the Cohere Embed 3 model [restrict](https://docs.cohere.com/v2/docs/rate-limits) you to 5 images per minute. 
```

Comment:
Oh wow 5/min! That's tiny. I think even the full one is like 40/min.

Maybe we could look at VoyageAI - their limits 2000 RPM for multimodal models.